mypy/messages.py

```diff
@@ -992,6 +997,28 @@ def pretty_or(args: List[str]) -> str:
     return ", ".join(quoted[:-1]) + ", or " + quoted[-1]
 
 
+def invariance_notes(notes: List[str], arg_type: Type, expected_type: Type) -> List[str]:
+    """Explain that the type is invariant and give notes for how to solve the issue."""
+    if isinstance(arg_type, Instance) and isinstance(expected_type, Instance):
```

Comment:
I proposed to place this at the call site of this function not at the definition site, to avoid unnecessary calls.
(Also this will save you 4 spaces of indent.)